tests/integration/test_runs.py

```diff
@@ -129,3 +130,36 @@ def test_create_run_and_save_model(
     finetuner_mocker.delete_experiment(experiment_name)
     experiments = finetuner_mocker.list_experiments()
     assert experiment_name not in [experiment.name for experiment in experiments]
+
+
+def test_create_synthesis_run_and_save_data(
+    finetuner_mocker, synthesis_query_data, synthesis_corpus_data
+):
+    import time
+
+    experiment_name = create_random_name()
+    finetuner_mocker.create_experiment(name=experiment_name)
+    run = finetuner_mocker.create_synthesis_run(
+        query_data=synthesis_query_data,
+        corpus_data=synthesis_corpus_data,
+        num_relations=3,
+        experiment_name=experiment_name,
+    )
+    status = run.status()[STATUS]
+    while status not in [FAILED, FINISHED]:
+        time.sleep(10)
+        status = run.status()[STATUS]
```

Comment:
yes in the CI it should time out at some point, but I think it is not good to rely only on that. Because, you might want to see if other tests still running and also because you might want to run those tests locally. Maybe replace it with a for loop if it really needs multiple tries.